packages/opentelemetry-exporter-collector/src/platform/node/CollectorExporter.ts

```diff
@@ -14,29 +14,34 @@
  * limitations under the License.
  */
 
-import * as protoLoader from '@grpc/proto-loader';
-import * as grpc from 'grpc';
-import * as path from 'path';
-import * as collectorTypes from '../../types';
-
 import { ReadableSpan } from '@opentelemetry/tracing';
+import * as grpc from 'grpc';
 import {
   CollectorExporterBase,
   CollectorExporterConfigBase,
 } from '../../CollectorExporterBase';
-import { CollectorExporterError } from '../../types';
-import { toCollectorExportTraceServiceRequest } from '../../transform';
+import * as collectorTypes from '../../types';
+import {
+  DEFAULT_COLLECTOR_URL_GRPC,
+  onInitWithGrpc,
+  sendSpansUsingGrpc,
+} from './utilWithGrpc';
+import {
+  DEFAULT_COLLECTOR_URL_JSON,
+  onInitWithJson,
+  sendSpansUsingJson,
+} from './utilWithJson';
 import { GRPCQueueItem, TraceServiceClient } from './types';
-import { removeProtocol } from './util';
-
-const DEFAULT_COLLECTOR_URL = 'localhost:55678';
 
 /**
  * Collector Exporter Config for Node
+ * headers will only work if useJson is set to true
  */
 export interface CollectorExporterConfig extends CollectorExporterConfigBase {
   credentials?: grpc.ChannelCredentials;
   metadata?: grpc.Metadata;
+  headers?: { [key: string]: string };
+  useJson?: boolean;
```

Comment:
I think it would make sense and would be clearer if we used an enum here, something like:
```ts
enum CollectorTransport {
   GRPC,
   HTTP_JSON
}
```

Reply:
changed